add_test.go

```diff
@@ -44,9 +45,163 @@ func TestAdd(t *testing.T) {
 		}
 
 		runF(t, clusters, f)
+
+		// for next test, this needs to be unpinned.
+		err = clusters[0].Unpin(ci)
+		if err != nil {
+			t.Fatal(err)
+		}
+		pinDelay()
+	})
+
+	t.Run("shard", func(t *testing.T) {
```

Comment:
It would be great if each of these tests could have a bit more explanation as to what they are testing as the testcase is hard to grok from the code.